src/GafferSceneUI/Inspector.cpp

```diff
@@ -301,6 +310,303 @@ void Inspector::editScopeInputChanged( const Gaffer::Plug *plug )
 	}
 }
 
+PathPtr Inspector::historyPath()
+{
+	return new Inspector::HistoryPath( this, new Context( *Context::current() ) );
+}
+
+//////////////////////////////////////////////////////////////////////////
+// HistoryPath
+//////////////////////////////////////////////////////////////////////////
+
+Inspector::HistoryPath::HistoryPath(
+	const InspectorPtr inspector,
+	ConstContextPtr context,
+	const std::string &path,
+	PathFilterPtr filter
+) :
+	Path( path, filter ),
+	m_inspector( inspector ),
+	m_context( context ),
+	m_nodeNameConnections()
+{
+	assert( m_inspector );
+
+	Context::Scope currentScope( m_context.get() );
+
+	SceneAlgo::History::ConstPtr history = m_inspector->history();
+
+	assert( history );
+	assert( history->scene );
+
+	m_scriptNode = history->scene->source<ScenePlug>()->ancestor<ScriptNode>();
+
+	assert( m_scriptNode );
+
+	if( !isLeaf() )
+	{
+		// These signals update the whole path, so we can reduce the number of signals
+		// by only adding them to the root.
+		m_inspectorDirtiedConnection = m_inspector->dirtiedSignal().connect(
+			boost::bind( &HistoryPath::inspectorDirtied, this, ::_1 )
+		);
+		m_pathChangedConnection = pathChangedSignal().connect(
+			boost::bind( &HistoryPath::pathChanged, this, ::_1 )
+		);
+	}
+
+	emitPathChanged();
```

Comment:
I was confused by this because I was thinking "if we're still constructing, there can be no signal observers". But I think what we're doing here is ensuring that our own `pathChanged()` slot is called so that we update the node connections?

Reply:
Yes, that's what I was going for. Alternatively I could call `updateNodeNameChangedSignals()` but I went with this on the chance that something else may be added to `pathChanged` in the future that could fall through the cracks if it wasn't clear to add it in the constructor too.